src/ashley/models.py

```diff
@@ -197,3 +212,50 @@ class Meta:
 
 # Default implementation of the AbstractLTIContext model
 LTIContext = model_factory(AbstractLTIContext)
+
+
+def image_directory_path(instance, filename):
+    """File will be uploaded to specific path"""
+    return "{0}_{1}/{2}/{3}".format(
+        instance.forum.id, instance.forum.slug, instance.poster.id, filename
+    )
+
+
+class AbstractUploadImage(DatedModel):
+    """Abstract model for images."""
+
+    file = models.ImageField(
+        upload_to=image_directory_path,
+        max_length=100,
+        blank=True,
+        validators=[
+            validate_upload_image_file_size,
+            validate_upload_image_allowed_type,
+        ],
+        null=True,
+    )
+    forum = models.ForeignKey(
+        "forum.Forum",
+        on_delete=models.CASCADE,
+        verbose_name=_("Forum"),
+        blank=True,
+        null=True,
+    )
+    poster = models.ForeignKey(
+        get_user_model(),
+        blank=True,
+        null=True,
```

Comment:
Why is this field optional?
For which use case?

Reply:
I don't want anything due to saving in database to block the edition or draft-js, saving the image in the database is not primary so blank or null by default seems to me to be advised

Reply:
I think we should have a minimal amount of confidence in our data model and our database.
Have you identified a case where this could cause problem ?

Reply:
I've always set it this way by principle so I wouldn't know but it's a constraint that i didn't see the point on dealing with, not to care was to me the good level for this task.